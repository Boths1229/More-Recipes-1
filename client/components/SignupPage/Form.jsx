import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  onFocus: PropTypes.func.isRequired,
  value: PropTypes.shape({
    usernameError: PropTypes.string,
    emailError: PropTypes.string,
    confirmPasswordError: PropTypes.string,
  }),
};

const defaultProps = {
  value: null
};

const Form = props => (
  <div className="container">
    <div className="pt-5" style={{ textAlign: 'center' }}>
      <form className="m-x-auto form card" onSubmit={props.handleSubmit}>
        <h1 className="signin-text">More-Recipes</h1>
        <div className="pb-4 pt-3 form-group">
          <input
            type="text"
            name="username"
            className="form-control"
            required
            onChange={props.onChange}
            onFocus={props.onFocus}
            placeholder="Username"
          />
          {
            props.value.usernameError &&
            <div className="alert alert-danger">
              {
                props.value.usernameError
              }
            </div>
          }
        </div>
        <div className="pb-4 form-group">
          <input
            type="email"
            name="email"
            className="form-control"
            required
            onFocus={props.onFocus}
            onChange={props.onChange}
            placeholder="Email"
          />
          {
            props.value.emailError &&
            <div className="alert alert-danger">
              {
                props.value.emailError
              }
            </div>
          }
        </div>
        <div className="pb-4 form-group">
          <input
            type="password"
            id="pass"
            name="password"
            pattern=".{7,12}"
            title="7 to 12 characters"
            className="form-control"
            required
            onFocus={props.onFocus}
            onChange={props.onChange}
            placeholder="Password"
          />
        </div>
        <div className="pb-4 form-group">
          <input
            type="password"
            name="confirmPassword"
            className="form-control"
            required
            onFocus={props.onFocus}
            onChange={props.onChange}
            placeholder="Confirm Password"
          />
          {
            props.value.confirmPasswordError &&
            <div className="alert alert-danger">
              {
                props.value.confirmPasswordError
              }
            </div>
          }
        </div>
        <button type="submit" className="btn signin-btn btn-lg" style={{ color: '#f2b43c' }}>
          Create Account
        </button>
        <div className="form-group">
          <p>
            Already have an account? <a href="/signin">Log in.</a>
          </p>
        </div>
      </form>
    </div>
  </div>
);
Form.propTypes = propTypes;
Form.defaultProps = defaultProps;

export default Form;
