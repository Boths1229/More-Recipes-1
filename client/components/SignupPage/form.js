import React from 'react';


/* eslint-disable no-unused-vars */
const Form = (props) => {
  return(
    <div className="container">
      <div className="pt-5" style={{textAlign:'center'}}>
        <form className="m-x-auto form card">
          <h1 className="signin-text">More-Recipes</h1>
          <div className="pb-4 pt-3 form-group">
            <input type="text" className="form-control" required placeholder="Username"/>
          </div>
          <div className="pb-4 form-group">
            <input type="email" className="form-control" required placeholder="Email"/>
          </div>  
          <div className="pb-4 form-group">
            <input type="password" className="form-control" required placeholder="Password"/>
          </div>
          <div className="pb-4 form-group">
            <input type="password" className="form-control" required placeholder="Confirm Password"/>
          </div>
          <button type="submit" className="btn signin-btn btn-lg" style={{color:'#f2b43c'}}>
            Create Account
          </button>
          <div className="form-group">
            <p>
              Already have an account? <a href="/signin">Log in.</a>
            </p>
          </div>
        </form> 
      </div>
    </div>
  );
  
};

export default Form;