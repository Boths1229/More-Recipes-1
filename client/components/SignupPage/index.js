import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import toastr from 'toastr';
import signUpAction  from '../../actions/authAction';
import Header from './header';
import Form from './form';
import Footer from '../common/footer';
import img from '../../public/images/banner-img-2.jpg';


/* eslint-disable no-unused-vars */
/* eslint-disable require-jsdoc */
class SignupPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password: '',
      confirmPassword: '',
      error: '',
      confirmPasswordError: '',
      usernameError: '',
      emailError: '',
    }
    this.onChange = this.onChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  componentWillMount() {
    document.body.style.backgroundImage = `url(${img}`;
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.actions.signUpAction(this.state)
      .then(() => {
        toastr.success('Account successfully created ');
        toastr.success('Proceed to login');
      })
      .catch(err => {
        const data = err.response.data;
        switch (data.error) {
          case 'This username already exist':
            this.setState({
              usernameError: data.error
            });
            break;
          case 'This email already exists':
            this.setState({
              emailError: data.error
            });
            break;
          case 'Password does not match':
            this.setState({
              confirmPasswordError: data.error
            });
            break
          case 'Username must be greater than 5':
            this.setState({
              usernameError: data.error
            });
            break
          default:
            this.setState({
              error: data.error
            });
        }
      });
  }

  onChange(event) {
    const name = event.target.name,
      value = event.target.value;
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  
  render() {
    return (
      <div>
        <Header/>
        <Form
          value={this.state}
          onChange={this.onChange}
          handleSubmit={this.handleSubmit}/>
        <Footer/>
      </div>         
    );
  }
}

function mapDispatchToProps(dispatch){
  return {
    actions: bindActionCreators({
      signUpAction
    }, dispatch)
  }
}

export default connect(null, mapDispatchToProps)(SignupPage);