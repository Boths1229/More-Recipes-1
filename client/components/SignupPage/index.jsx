/* eslint-disable no-unused-vars */

import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import toastr from 'toastr';
import { signUpAction } from '../../actions/authAction';
import Header from './header.jsx';
import Form from './form.jsx';
import Footer from '../common/footer.jsx';
import img from '../../public/images/banner-img-2.jpg';


/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
/* eslint-disable require-jsdoc */
class SignupPage extends React.Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      username: '',
      email: '',
      password: '',
      confirmPassword: '',
      error: '',
      confirmPasswordError: '',
      usernameError: '',
      emailError: '',
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onFocus = this.onFocus.bind(this);
  }

  /* eslint-disable class-methods-use-this */
  componentWillMount() {
    document.body.style.backgroundImage = `url(${img}`;
  }

  onFocus(event) {    
    const { name } = event.target;
    switch (name) {
      case 'username':
        this.setState({ usernameError: '' });
        break;
      case 'email':
        this.setState({ emailError: '' });
        break;
      case 'confirmPassword':
        this.setState({ confirmPasswordError: '', });
        break;
      default:
        return this.state;
    }
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.actions.signUpAction(this.state)
      .then(() => {
        toastr.success('Account successfully created ');
        this.context.router.push('/signin');
      })
      .catch((err) => {
        const { data } = err.response;
        switch (data.error) {
          case 'This username already exist':
            this.setState({
              usernameError: data.error
            });
            break;
          case 'This email already exists':
            this.setState({
              emailError: data.error
            });
            break;
          case 'Password does not match':
            this.setState({
              confirmPasswordError: data.error
            });
            break;
          case 'Username must be greater than 5':
            this.setState({
              usernameError: data.error
            });
            break;
          default:
            this.setState({
              error: data.error
            });
        }
      });
  }

  onChange(event) {
    const { name } = event.target,
      { value } = event.target;
    this.setState({
      [name]: value
    });
  }

  

  render() {
    return (
      <div>
        <Header/>
        <Form
          value={this.state}
          onFocus={this.onFocus}
          onChange={this.onChange}
          handleSubmit={this.handleSubmit}/>
        <Footer/>
      </div>
    );
  }
}

SignupPage.contextTypes = {
  router: PropTypes.object
};

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      signUpAction
    }, dispatch)
  };
}

export default connect(null, mapDispatchToProps)(SignupPage);
