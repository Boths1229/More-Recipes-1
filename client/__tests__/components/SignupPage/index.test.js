import React from 'react';
import { mount } from 'enzyme';
import { SignupPage } from '../../../components/SignupPage/index';


describe('SignupPage component', () => {
  let wrapper;

  beforeEach(() => {
    const props = {
      actions: {
        resetUserError: () => {},
        signUpAction: () => {},
      }
    };
    wrapper = mount(<SignupPage {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should set username state when username input value changes', () => {
    const event = {
      target: { name: 'username', value: '' }
    };

    event.target.value = 'tywo';
    wrapper.find('input').at(0).simulate('change', event);
    expect(wrapper.state('username')).toBe('tywo');
  });

  it('should set usernameError when username input box is focused on', () => {
    wrapper.setState({ usernameError: 'user already exist' });

    const event = {
      target: { name: 'username', value: '' }
    };
    wrapper.find('input').at(0).simulate('focus', event);
    expect(wrapper.state('usernameError')).toBe('');
  });

  it('should set emailError when email input box is focused on', () => {
    wrapper.setState({ emailError: 'email already exist' });

    const event = {
      target: { name: 'email', value: '' }
    };
    wrapper.find('input').at(1).simulate('focus', event);
    expect(wrapper.state('emailError')).toBe('');
  });

  it('should set confirmPasswordError when confirmPassword input box is focused on', () => {
    wrapper.setState({ confirmPasswordError: 'password does not match' });

    const event = {
      target: { name: 'confirmPassword', value: '' }
    };
    wrapper.find('input').at(3).simulate('focus', event);
    expect(wrapper.state('confirmPasswordError')).toBe('');
  });

  it('should return state if none of the input is on focused', () => {
    const event = {
      target: { name: '', value: '' }
    };
    wrapper.find('input').at(3).simulate('focus', event);
    expect(wrapper.state('confirmPasswordError')).toBe('');
  });

  it('should set username Error when it recieves error props with `This username already exist`', () => {
    wrapper.setProps({ error: 'This username already exist' });
    expect(wrapper.state('usernameError')).toBe('This username already exist');
  });

  it('should set username Error when it recieves error props with `Username is required`', () => {
    wrapper.setProps({ error: 'Username is required' });
    expect(wrapper.state('usernameError')).toBe('Username is required');
  });

  it('should set username Error when it recieves error props with `Username must be greater than 5`', () => {
    wrapper.setProps({ error: 'Username must be greater than 5' });
    expect(wrapper.state('usernameError')).toBe('Username must be greater than 5');
  });

  it('should set email Error when it recieves error props with `This email already exists`', () => {
    wrapper.setProps({ error: 'This email already exists' });
    expect(wrapper.state('emailError')).toBe('This email already exists');
  });

  it('should set confirmPassword Error when it recieves error props with `Password does not match`', () => {
    wrapper.setProps({ error: 'Password does not match' });
    expect(wrapper.state('confirmPasswordError')).toBe('Password does not match');
  });

  it('should set error when error props is not specified', () => {
    wrapper.setProps({ error: 'server error' });
    expect(wrapper.state('error')).toBe('server error');
  });

  it('should register user when form is submitted', () => {
    const action = wrapper.instance();
    const signup = jest.spyOn(wrapper.instance(), 'handleSubmit');
    action.handleSubmit({ preventDefault: () => {} });
    expect(signup).toBeCalled();
  });
});
