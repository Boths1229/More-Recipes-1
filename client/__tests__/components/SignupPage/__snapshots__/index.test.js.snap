// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupPage component should render correctly 1`] = `
<SignupPage
  actions={
    Object {
      "resetUserError": [Function],
      "signUpAction": [Function],
    }
  }
  error={null}
  response={null}
>
  <div>
    <Header>
      <header>
        <nav
          className="navbar navbar-expand-lg navbar-dark"
          style={
            Object {
              "backgroundColor": "#2b3034",
            }
          }
        >
          <a
            className="navbar-brand"
            href="/"
          >
            More-Recipes
          </a>
          <button
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-target="#navbarSupportedContent"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>
          <div
            className="collapse navbar-collapse"
            id="navbarSupportedContent"
          >
            <ul
              className="navbar-nav ml-auto"
            >
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/signin"
                >
                  <a
                    className="nav-link"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Login
                  </a>
                </Link>
              </li>
            </ul>
          </div>
        </nav>
      </header>
    </Header>
    <Form
      handleSubmit={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value={
        Object {
          "confirmPassword": "",
          "confirmPasswordError": "",
          "email": "",
          "emailError": "",
          "error": "",
          "password": "",
          "username": "",
          "usernameError": "",
        }
      }
    >
      <div
        className="container"
      >
        <div
          className="pt-5"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <form
            className="m-x-auto form card"
            onSubmit={[Function]}
          >
            <h1
              className="signin-text"
            >
              More-Recipes
            </h1>
            <div
              className="pb-4 pt-3 form-group"
            >
              <input
                className="form-control"
                name="username"
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Username"
                required={true}
                type="text"
              />
            </div>
            <div
              className="pb-4 form-group"
            >
              <input
                className="form-control"
                name="email"
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Email"
                required={true}
                type="email"
              />
            </div>
            <div
              className="pb-4 form-group"
            >
              <input
                className="form-control"
                id="pass"
                name="password"
                onChange={[Function]}
                onFocus={[Function]}
                pattern=".{7,12}"
                placeholder="Password"
                required={true}
                title="7 to 12 characters"
                type="password"
              />
            </div>
            <div
              className="pb-4 form-group"
            >
              <input
                className="form-control"
                name="confirmPassword"
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Confirm Password"
                required={true}
                type="password"
              />
            </div>
            <button
              className="btn signin-btn btn-lg"
              style={
                Object {
                  "color": "#f2b43c",
                }
              }
              type="submit"
            >
              Create Account
            </button>
            <div
              className="form-group"
            >
              <p>
                Already have an account? 
                <a
                  href="/signin"
                >
                  Log in.
                </a>
              </p>
            </div>
          </form>
        </div>
      </div>
    </Form>
    <Footer>
      <footer
        className="footer-section"
        id="footer-below"
      >
        <div
          className="container"
        >
          <strong
            className="copyright"
          >
            More-Recipes Â© 2018, All Rights Reserved, Design & Developed By: SannyTee
          </strong>
        </div>
      </footer>
    </Footer>
  </div>
</SignupPage>
`;
